### PR TITLE
AUTH-323: Add etcd and etcd metrics certs, add etcd metrics proxy

### PR DESCRIPTION
**Pull request by ibihim**
**What this PR does / why we need it**:

What:
- Decouple etcd trust chain from root CA.
- Use etcd proxy for metrics.
- Add etcd metrics trust chain.
- Add distinct certificates for etcd proxy. 

Why:
- Create distinct trust zones.
- Use distinct certs for metrics scraping.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/manifests/pki.go [hunk 1]
```diff
@@ -23,69 +32,77 @@ func ClusterSignerCASecret(ns string) *corev1.Secret {
 	}
 }
 
-func CombinedCAConfigMap(ns string) *corev1.ConfigMap {
```
Comment (ibihim) [inline control-plane-operator/controllers/hostedcontrolplane/manifests/pki.go:26]: `git diff` fails here. Isn't removed. Is moved down.

File: control-plane-operator/controllers/hostedcontrolplane/manifests/pki.go [hunk 2]
```diff
@@ -23,69 +32,77 @@ func ClusterSignerCASecret(ns string) *corev1.Secret {
...
+	return &corev1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "metrics-client",
+			Name:      "etcd-metrics-ca",
 			Namespace: ns,
 		},
 	}
 }
 
-func UserCAConfigMap(ns string) *corev1.ConfigMap {
```
Comment (ibihim) [inline control-plane-operator/controllers/hostedcontrolplane/manifests/pki.go:44]: `git diff` fails here. Isn't removed. Is moved down.

File: control-plane-operator/controllers/hostedcontrolplane/manifests/pki.go [hunk 3]
```diff
@@ -23,69 +32,77 @@ func ClusterSignerCASecret(ns string) *corev1.Secret {
...
 	return &corev1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "user-ca-bundle",
+			Name:      "combined-ca",
 			Namespace: ns,
 		},
 	}
 }
 
-func EtcdClientSecret(ns string) *corev1.Secret {
```
Comment (ibihim) [inline control-plane-operator/controllers/hostedcontrolplane/manifests/pki.go:53]: `git diff` fails here. Isn't removed. Is moved down.

File: control-plane-operator/controllers/hostedcontrolplane/manifests/pki.go [hunk 4]
```diff
@@ -23,69 +32,77 @@ func ClusterSignerCASecret(ns string) *corev1.Secret {
...
 	return &corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "etcd-client-tls",
+			Name:      "metrics-client",
 			Namespace: ns,
 		},
 	}
 }
 
-func EtcdServerSecret(ns string) *corev1.Secret {
```
Comment (ibihim) [inline control-plane-operator/controllers/hostedcontrolplane/manifests/pki.go:62]: `git diff` fails here. Isn't removed. Is moved down.

File: control-plane-operator/controllers/hostedcontrolplane/etcd/reconcile.go [hunk 5]
```diff
@@ -220,15 +252,15 @@ func buildEtcdContainer(p *EtcdParams, namespace string) func(c *corev1.Containe
 			},
 			{
 				Name:          "metrics",
-				ContainerPort: 2381,
+				ContainerPort: 2382,
 				Protocol:      corev1.ProtocolTCP,
 			},
```
Comment (stlaz) [inline control-plane-operator/controllers/hostedcontrolplane/etcd/reconcile.go:257]: this port should not be published

**Review by stlaz — CHANGES_REQUESTED** [review]

**Pull request merged**